pkg/ddl/column.go

```diff
@@ -1948,7 +1948,9 @@ func modifyColsFromNull2NotNull(w *worker, dbInfo *model.DBInfo, tblInfo *model.
 func generateOriginDefaultValue(col *model.ColumnInfo, ctx sessionctx.Context) (any, error) {
 	var err error
 	odValue := col.GetDefaultValue()
-	if odValue == nil && mysql.HasNotNullFlag(col.GetFlag()) {
+	if odValue == nil && mysql.HasNotNullFlag(col.GetFlag()) ||
+		// It's for drop column and modify column.
+		(col.DefaultIsExpr && odValue != strings.ToUpper(ast.CurrentTimestamp) && ctx == nil) {
```

Comment:
Why strings.ToUpper here？

Reply:
Here current_timestamp is `CURRENT_TIMESTAMP`, in addition, it also refers to the line1972